[BUG] Error: Cannot find module '@shopware-pwa/cms-base/module'

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

I have installed 

```
{
  "name": "demo-store",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "nuxt dev",
    "build": "nuxt build",
    "start": "nuxt start",
    "lint": "eslint . --fix",
    "typecheck": "nuxi typecheck",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "analyze": "nuxt analyze",
    "format": "prettier --write --list-different ."
  },
  "dependencies": {
    "@iconify/tools": "^4.0.7",
    "@nuxt/image": "1.8.1",
    "@nuxtjs/google-fonts": "^3.2.0",
    "@nuxtjs/tailwindcss": "^6.12.2",
    "@shopware-pwa/composables-next": "^1.3.0",
    "@shopware-pwa/helpers-next": "^1.1.0",
    "@shopware-pwa/nuxt3-module": "^1.0.5",
    "@shopware-pwa/shopware-6-client": "^1.6.1",
    "@shopware-pwa/types": "^1.0.2",
    "@shopware/api-client": "^1.0.2",
    "@vuelidate/core": "^2.0.3",
    "@vuelidate/validators": "^2.0.4",
    "@vueuse/nuxt": "^11.1.0",
    "leaflet-gesture-handling": "^1.2.2",
    "nuxt-lazy-load": "^3.0.4",
    "nuxt-multi-cache": "^3.3.3",
    "nuxt-svgo": "^4.0.6",
    "nuxt-swiper": "^1.2.2",
    "requrl": "^3.0.2",
    "sitemap": "^8.0.0",
    "swiper": "^11.1.14",
    "vue": "^3.5.12",
    "vue-leaflet-markercluster": "^0.6.1"
  },
  "devDependencies": {
    "@iconify-json/carbon": "^1.2.3",
    "@nuxt/devtools": "^1.6.0",
    "@nuxt/eslint": "^0.6.0",
    "@nuxtjs/i18n": "^8.5.5",
    "@shopware-pwa/cms-base": "^1.0.5",
    "@zadigetvoltaire/nuxt-gtm": "^0.0.13",
    "@zoom-avant/nuxt": "^0.7.1",
    "eslint": "^9.13.0",
    "maska": "^3.0.3",
    "nuxt": "^3.13.2",
    "nuxt-security": "^2.0.0",
    "nuxt3-leaflet": "^1.0.13",
    "prettier": "^3.3.3",
    "prettier-plugin-tailwindcss": "^0.6.8",
    "sass": "^1.80.3",
    "typescript": "^5.6.3"
  },
  "engines": {
    "node": "^20.x"
  }
}
```

This are my modules in the `nuxt.config.ts`

```
extends: ['@shopware-pwa/composables-next/nuxt-layer'],

  modules: [
    'nuxt3-leaflet',
    '@vueuse/nuxt',
    '@shopware-pwa/nuxt3-module',
    '@shopware-pwa/cms-base',
    '@nuxtjs/google-fonts',
    '@nuxtjs/tailwindcss',
    '@nuxtjs/i18n',
    'nuxt-lazy-load',
    'nuxt-security',
    '@zadigetvoltaire/nuxt-gtm',
    '@zoom-avant/nuxt',
    'nuxt-svgo',
    '@nuxt/image',
    '@nuxt/eslint',
  ],
```


when I start my server with `pnpm dev`, I get the following error:


```
 ERROR  Error while requiring module [object Object]: Error: Cannot find module '@shopware-pwa/cms-base/module'                                      17:03:11  
Require stack:
- C:\Users\developer\development\nuxt\SN.Shopware.Headless\index.js


 ERROR  Cannot start nuxt:  Cannot find module '@shopware-pwa/cms-base/module'                                                                       17:03:11  
Require stack:
- C:\Users\developer\development\nuxt\SN.Shopware.Headless\index.js
```

I tried deleting and reinstalling the lock file, `pnpm nuxi cleanup`, reinstalling `node_modules` etc, but unfortunately I have continuously this error

### Expected Behavior

No error when starting the Nuxt app

### Steps To Reproduce

1. Install the latest `@shopware-pwa/cms-base` and Nuxt version
2. Add it to the modules section of the nuxt config
3. Try to start the server

### Environment

- OS:windows
- Node:20
- pnpm:9.12.1


### Anything else?

Thank you and kind regards

--- a/.husky/pre-commit
+++ b/.husky/pre-commit
@@ -1,3 +1,4 @@
 pnpm run lint
+pnpm generateAutomd
 prettier $(git diff --cached --name-only --diff-filter=ACMR | sed 's| |\\ |g') --write --ignore-unknown
 git update-index --again

--- a/apps/docs/src/getting-started/features/broadcasting.md
+++ b/apps/docs/src/getting-started/features/broadcasting.md
@@ -46,7 +46,7 @@ This way:
 - cart data
   are synchronized between tabs.
 
-<!-- automd:file src="../../../../../templates/vue-demo-store/composables/useBroadcastChannelSync.ts" code -->
+<!-- automd:file src="templates/vue-demo-store/composables/useBroadcastChannelSync.ts" code -->
 
 ```ts [useBroadcastChannelSync.ts]
 import type { Schemas } from "#shopware";

--- a/apps/docs/src/getting-started/templates/custom-vue-project.md
+++ b/apps/docs/src/getting-started/templates/custom-vue-project.md
@@ -38,9 +38,26 @@ You can read more about CMS pages here:
 
 <PageRef page="../cms/content-pages.html" title="Create content pages" sub="Render a content page using components" />
 
-```bash
-pnpm add @shopware-pwa/cms-base
+<!-- automd:pm-install name="@shopware-pwa/cms-base" dev -->
+
+```sh
+# ✨ Auto-detect
+npx nypm install -D @shopware-pwa/cms-base
+
+# npm
+npm install -D @shopware-pwa/cms-base
+
+# yarn
+yarn add -D @shopware-pwa/cms-base
+
+# pnpm
+pnpm install -D @shopware-pwa/cms-base
+
+# bun
+bun install -D @shopware-pwa/cms-base
 ```
+
+<!-- /automd -->
 
 ## Configure API client
 

--- a/package.json
+++ b/package.json
@@ -27,6 +27,7 @@
     "prepare": "husky",
     "generate-types": "turbo run generate-types",
     "generateDependencyChangelog": "esno ./scripts/generateDependencyChangelog.ts",
+    "generateAutomd": "esno ./scripts/generateAutomd.ts --update",
     "docs:env": "[ -d \"../developer-portal\" ] && ../developer-portal/docs-cli.cjs pull || (git clone git@github.com:shopware/developer-portal.git ../developer-portal && pnpm i -C ../developer-portal)",
     "docs:link": "../developer-portal/docs-cli.cjs link --src apps/docs/src --dst frontends --symlink --keep",
     "docs:preview": "../developer-portal/docs-cli.cjs preview",
@@ -37,6 +38,7 @@
     "@changesets/changelog-github": "0.5.0",
     "@changesets/cli": "2.27.9",
     "@types/changelog-parser": "2.8.4",
+    "automd": "^0.3.7",
     "changelog-parser": "3.0.1",
     "dotenv-cli": "7.4.2",
     "esno": "4.8.0",
@@ -101,7 +103,8 @@
       ]
     },
     "patchedDependencies": {
-      "@vueuse/core@11.1.0": "patches/@vueuse__core@11.1.0.patch"
+      "@vueuse/core@11.1.0": "patches/@vueuse__core@11.1.0.patch",
+      "automd": "patches/automd.patch"
     }
   }
 }

--- a/packages/api-gen/README.md
+++ b/packages/api-gen/README.md
@@ -43,7 +43,7 @@ If your instance contains inacurate or outdated OpenAPI specification, you can o
 
 Example of overrides file:
 
-<!-- automd:file src="./tests/snapshots-override/simpleOverride.example.ts" code -->
+<!-- automd:file src="packages/api-gen/tests/snapshots-override/simpleOverride.example.ts" code -->
 
 ```ts [simpleOverride.example.ts]
 import { components as mainComponents } from "./storeApiTypes";

--- a/packages/cms-base/README.md
+++ b/packages/cms-base/README.md
@@ -5,7 +5,7 @@
 [![](https://img.shields.io/github/issues/shopware/frontends/cms-base?label=cms-base%20issues&logo=github)](https://github.com/shopware/frontends/issues?q=is%3Aopen+is%3Aissue+label%3Acms-base)
 [![](https://img.shields.io/github/license/shopware/frontends?color=blue)](#)
 
-Nuxt [module](https://nuxt.com/docs/guide/going-further/modules) that provides an implementation of all CMS components in Shopware [based on utility-classes](https://frontends.shopware.com/framework/styling.html) using unocss/Tailwind.css syntax.
+Nuxt [layer](https://nuxt.com/docs/getting-started/layers) that provides an implementation of all CMS components in Shopware [based on utility-classes](https://frontends.shopware.com/framework/styling.html) using atomic css syntax (UnoCss / Tailwind).
 
 It is useful for projects that want to use the CMS components but design their own layout.
 
@@ -19,27 +19,56 @@ It is useful for projects that want to use the CMS components but design their o
 
 Install npm package:
 
-```bash
-# Using pnpm
-pnpm add -D @shopware-pwa/cms-base
+<!-- automd:pm-install name="@shopware-pwa/cms-base" dev -->
 
-# Using yarn
-yarn add --dev @shopware-pwa/cms-base
+```sh
+# ✨ Auto-detect
+npx nypm install -D @shopware-pwa/cms-base
 
-# Using npm
-npm i @shopware-pwa/cms-base --save-dev
+# npm
+npm install -D @shopware-pwa/cms-base
+
+# yarn
+yarn add -D @shopware-pwa/cms-base
+
+# pnpm
+pnpm install -D @shopware-pwa/cms-base
+
+# bun
+bun install -D @shopware-pwa/cms-base
 ```
 
-Then, register the module by editing `nuxt.config.js` or (`.ts`) file:
+<!-- /automd -->
 
-```js
-/* nuxt.config.ts */
+Then, register the Nuxt layer in `nuxt.config.ts` file:
 
+<!-- automd:file src="templates/vue-blank/nuxt.config.ts" code -->
+
+```ts [nuxt.config.ts]
+// https://v3.nuxtjs.org/api/configuration/nuxt.config
 export default defineNuxtConfig({
-  /* ... */
-+ modules: [/* ... */, "@shopware-pwa/cms-base"],
+  extends: [
+    "@shopware-pwa/composables-next/nuxt-layer",
+    "@shopware-pwa/cms-base",
+  ],
+  shopware: {
+    endpoint: "https://demo-frontends.shopware.store/store-api/",
+    accessToken: "SWSCBHFSNTVMAWNZDNFKSHLAYW",
+  },
+  modules: ["@shopware-pwa/nuxt3-module"],
+  /**
+   * Commented because of the StackBlitz error
+   * Issue: https://github.com/shopware/frontends/issues/88
+   */
+  typescript: {
+    // typeCheck: true,
+    strict: true,
+  },
+  telemetry: false,
 });
 ```
+
+<!-- /automd -->
 
 ### ⚠️ `<RouterLink/>` components used
 

--- a/packages/composables/README.md
+++ b/packages/composables/README.md
@@ -104,7 +104,7 @@ npm i js-cookie
 
 Let's get back to the step where the `api-client` was initialized:
 
-<!-- automd:file src="../../examples/b2b-quote-management/src/apiClient.ts" code -->
+<!-- automd:file src="examples/b2b-quote-management/src/apiClient.ts" code -->
 
 ```ts [apiClient.ts]
 import { createAPIClient } from "@shopware/api-client";

--- a/patches/automd.patch
+++ b/patches/automd.patch
@@ -1,0 +1,24 @@
+diff --git a/dist/index.cjs b/dist/index.cjs
+index 754e3ac52c21f200cc8520bf77337947c57e7923..fd6e01f6376c2bdfa9a6693d191b1ec9cc4f1936 100644
+--- a/dist/index.cjs
++++ b/dist/index.cjs
+@@ -489,6 +489,7 @@ const file = defineGenerator({
+   async generate({ args, config, url }) {
+     const fullPath = resolvePath(args.src, { url, dir: config.dir });
+     let contents = await promises.readFile(fullPath, "utf8");
++    contents = contents.trim();
+     if (args.code) {
+       contents = mdbox.md.codeBlock(
+         contents,
+diff --git a/dist/index.mjs b/dist/index.mjs
+index 5a800f1256f2ca858bd64a8e6988a86f82429064..fbd3d61f5fae9e9fc1735ab3c83f034c854fe790 100644
+--- a/dist/index.mjs
++++ b/dist/index.mjs
+@@ -483,6 +483,7 @@ const file = defineGenerator({
+   async generate({ args, config, url }) {
+     const fullPath = resolvePath(args.src, { url, dir: config.dir });
+     let contents = await readFile(fullPath, "utf8");
++    contents = contents.trim();
+     if (args.code) {
+       contents = md.codeBlock(
+         contents,

--- a/scripts/generateAutomd.ts
+++ b/scripts/generateAutomd.ts
@@ -1,0 +1,111 @@
+import * as fs from "fs";
+import * as path from "path";
+import { transform, TransformResult } from "automd";
+
+/**
+ * This script will search for all markdown files in the project and check if they contain "automd:" term.
+ * If they do, it will run the automd transformation and update the file if needed.
+ */
+
+const stats: {
+  changedFiled: number;
+  filesWithIssues: Array<TransformResult & { path: string }>;
+} = {
+  changedFiled: 0,
+  filesWithIssues: [],
+};
+// read args, see if flag --update is present
+const shouldUpdate = process.argv.includes("--update");
+
+async function run() {
+  const start = performance.now();
+
+  await findInFiles({
+    term: "automd:",
+    rootPath: process.cwd(),
+    fileTypes: ["md"],
+    // exclude node_modules
+    exclude: ["node_modules"],
+  });
+
+  // display issues
+  if (stats.filesWithIssues.length) {
+    console.log(`Found ${stats.filesWithIssues.length} files with issues:`);
+    for (const resultWithIssue of stats.filesWithIssues) {
+      console.log(`- ${resultWithIssue.path}`);
+      for (const issue of resultWithIssue.updates) {
+        for (const issueMessage of issue.result.issues || []) {
+          console.log(`  - ${issueMessage}`);
+        }
+      }
+    }
+  }
+
+  // display changhed files:
+  console.log(`\nChanged ${stats.changedFiled} files.`);
+  const end = performance.now();
+  console.log(`Time: ${Math.round(end - start)}ms`);
+}
+
+run();
+
+async function findInFiles({
+  term,
+  rootPath,
+  fileTypes,
+  exclude,
+}: {
+  term: string;
+  rootPath: string;
+  fileTypes: string[];
+  exclude: string[];
+}): Promise<string[]> {
+  const filePaths: string[] = [];
+
+  async function searchDirectory(directory: string) {
+    const files = await fs.promises.readdir(directory, { withFileTypes: true });
+
+    for (const file of files) {
+      const fullPath = path.join(directory, file.name);
+
+      if (file.isDirectory()) {
+        if (!exclude.includes(file.name)) {
+          await searchDirectory(fullPath);
+        }
+      } else if (fileTypes.includes(path.extname(file.name).substring(1))) {
+        const content = await fs.promises.readFile(fullPath, "utf8");
+        if (content.includes(term)) {
+          filePaths.push(fullPath);
+          const updatedContent = await transform(content, {
+            dir: process.cwd(),
+          });
+          if (updatedContent.hasIssues) {
+            stats.filesWithIssues.push({
+              ...updatedContent,
+              path: fullPath,
+            });
+          } else {
+            if (updatedContent.hasChanged) {
+              if (shouldUpdate) {
+                await fs.promises.writeFile(
+                  fullPath,
+                  updatedContent.contents,
+                  "utf8",
+                );
+                stats.changedFiled++;
+              } else {
+                // if --update flag is not present, then it's just a check so we throw an error
+                throw new Error(
+                  `File ${fullPath} should be updated with "automd" script. Run with "pnpm generateAutomd" to fix.`,
+                );
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  await searchDirectory(rootPath);
+  return filePaths;
+}

--- a/templates/astro/README.md
+++ b/templates/astro/README.md
@@ -17,7 +17,7 @@ Go to [Documentation > Requirements](https://frontends.shopware.com/framework/re
 
 In order to have a different API connected to the app, change two lines of code in [./src/entrypoints/\_shopware.ts](./src/entrypoints/_shopware.ts):
 
-<!-- automd:file src="./src/entrypoints/_shopware.ts" code -->
+<!-- automd:file src="templates/astro/src/entrypoints/_shopware.ts" code -->
 
 ```ts [_shopware.ts]
 import type { App } from "vue";

